elkodon/src/compiletests.rs

```diff
@@ -1,21 +1,43 @@
 /// ```compile_fail
 /// use elkodon::prelude::*;
-/// # fn main() -> Result<(), Box<dyn std::error::Error>> {
-/// # let service_name = ServiceName::new(b"My/Funk/ServiceName").unwrap();
-/// #
-/// # let service = zero_copy::Service::new(&service_name)
-/// #     .publish_subscribe()
-/// #     .open_or_create::<u64>()?;
-/// #
-/// # let publisher = service.publisher().create()?;
-///
-/// let mut sample = publisher.loan()?;
+/// fn main() -> Result<(), Box<dyn std::error::Error>> {
+/// let service_name = ServiceName::new(b"My/Funk/ServiceName").unwrap();
+///
+/// let service = zero_copy::Service::new(&service_name)
+///     .publish_subscribe()
+///     .open_or_create::<u64>()?;
+///
+/// let publisher = service.publisher().create()?;
+///
+/// let mut sample = publisher.loan_uninit()?;
 /// sample.payload_mut().write(1234);
 ///
 /// publisher.send(sample)?; // should fail to compile
 ///
-/// # Ok(())
-/// # }
+/// Ok(())
+/// }
 /// ```
 #[cfg(doctest)]
 fn sending_uninitialized_sample_fails_to_compile() {}
+
+/// ```compile_fail
+/// use elkodon::prelude::*;
+///
+/// struct Wrapper(u64);
+///
+/// fn main() -> Result<(), Box<dyn std::error::Error>> {
+/// let service_name = ServiceName::new(b"My/Funk/ServiceName").unwrap();
+///
+/// let service = zero_copy::Service::new(&service_name)
+///     .publish_subscribe()
+///     .open_or_create::<Wrapper>()?;
+///
+/// let publisher = service.publisher().create()?;
+///
+/// let sample = publisher.loan()?; // should fail to compile
```

Comment:
Same here, could you please explain why it fails and what the user has to do to solve the compilation problem.